x-pack/plugins/security_solution/cypress/e2e/detection_alerts/alerts_charts.cy.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getNewRule } from '../../objects/rule';
+import { ALERTS_HISTOGRAM_LEGEND } from '../../screens/alerts';
+import { waitForAlerts, selectAlertsHistogram } from '../../tasks/alerts';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { login, visit } from '../../tasks/login';
+import { ALERTS_URL } from '../../urls/navigation';
+import { GLOBAL_SEARCH_BAR_FILTER_ITEM } from '../../screens/search_bar';
+import { HOVER_ACTIONS, TIMELINE_DATA_PROVIDERS_CONTAINER } from '../../screens/timeline';
+import { closeTimelineUsingCloseButton } from '../../tasks/security_main';
+import { openActiveTimeline } from '../../tasks/timeline';
+
+describe('Alerts page charts', () => {
+  before(() => {
+    cleanKibana();
+    login();
+    createCustomRuleEnabled(getNewRule(), 'new custom rule');
+  });
+
+  context('Histogram legend hover actions', () => {
+    beforeEach(() => {
+      visit(ALERTS_URL);
+      waitForAlerts();
+      selectAlertsHistogram();
+    });
+
+    it('Filter For', () => {
+      cy.get(ALERTS_HISTOGRAM_LEGEND)
+        .eq(0)
+        .trigger('mouseover')
+        .then(($el) => {
+          const textValue = $el.text();
+          cy.get(HOVER_ACTIONS.FILTER_FOR).should('exist');
+          cy.get(HOVER_ACTIONS.FILTER_FOR).trigger('click', { force: true });
+          cy.get(GLOBAL_SEARCH_BAR_FILTER_ITEM).should(
+            'contain.text',
+            `kibana.alert.rule.name: ${textValue}`
+          );
+        });
+    });
+
+    it('Filter Out', () => {
```

Comment:
while it's nice to have separate tests most of the time, one could argue that this one and the previous could be combined into a `should add filter in and out to KQL bar from the chart legend`.
That would save a bit of time on the CI, and these functionalities are extremely related in code and in user interation.

Reply:
replaced `beforeEach` with `before` (see below), to your point, some of the set up does not need to be run for every test. that way it should not incur more time even if we break it out to multiple tests